client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/Role.java

```diff
@@ -0,0 +1,285 @@
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(clusterPrivileges, globalApplicationPrivileges, indicesPrivileges, applicationResourcePrivileges,
+                runAsPrivilege, metadata);
+    }
+
+    @Override
+    public String toString() {
+        try {
+            return XContentHelper.toXContent(this, XContentType.JSON, true).utf8ToString();
+        } catch (IOException e) {
+            throw new RuntimeException("Unexpected", e);
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (false == clusterPrivileges.isEmpty()) {
+            builder.field(CLUSTER.getPreferredName(), clusterPrivileges);
+        }
+        if (null != globalApplicationPrivileges) {
+            builder.field(GLOBAL.getPreferredName(), globalApplicationPrivileges);
+        }
+        if (false == indicesPrivileges.isEmpty()) {
+            builder.field(INDICES.getPreferredName(), indicesPrivileges);
+        }
+        if (false == applicationResourcePrivileges.isEmpty()) {
+            builder.field(APPLICATIONS.getPreferredName(), applicationResourcePrivileges);
+        }
+        if (false == runAsPrivilege.isEmpty()) {
+            builder.field(RUN_AS.getPreferredName(), runAsPrivilege);
+        }
+        if (false == metadata.isEmpty()) {
+            builder.field(METADATA.getPreferredName(), metadata);
+        }
+        return builder.endObject();
+    }
+
+    public static Role fromXContent(XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static final class Builder {
+
+        private @Nullable Collection<String> clusterPrivileges = null;
+        private @Nullable GlobalPrivileges globalApplicationPrivileges = null;
+        private @Nullable Collection<IndicesPrivileges> indicesPrivileges = null;
+        private @Nullable Collection<ApplicationResourcePrivileges> applicationResourcePrivileges = null;
+        private @Nullable Collection<String> runAsPrivilege = null;
+        private @Nullable Map<String, Object> metadata = null;
+
+        private Builder() {
+        }
+
+        public Builder clusterPrivileges(String... clusterPrivileges) {
+            return clusterPrivileges(Arrays
+                    .asList(Objects.requireNonNull(clusterPrivileges, "Cluster privileges cannot be null. Pass an empty array instead.")));
+        }
+
+        public Builder clusterPrivileges(Collection<String> clusterPrivileges) {
+            this.clusterPrivileges = Objects.requireNonNull(clusterPrivileges,
+                    "Cluster privileges cannot be null. Pass an empty collection instead.");
+            return this;
+        }
+
+        public Builder manageApplicationPrivileges(GlobalPrivileges globalApplicationPrivileges) {
```

Comment:
This method name no longer matches the object model (argument type)

Reply:
👍